crates/biome_js_analyze/src/assists/source/use_sorted_properties.rs

```diff
@@ -0,0 +1,225 @@
+use biome_analyze::ActionCategory;
+use biome_analyze::SourceActionKind;
+use biome_rowan::Direction;
+use std::borrow::Cow;
+use std::cmp::Ordering;
+
+use biome_analyze::{context::RuleContext, declare_source_rule, Ast, Rule, RuleAction};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::{AnyJsObjectMember, AnyJsObjectMemberName, JsObjectMemberList};
+use biome_rowan::{AstNode, AstSeparatedList, BatchMutationExt, SyntaxResult};
+use serde::{Deserialize, Serialize};
+
+use crate::JsRuleAction;
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize, Deserializable)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields, default)]
+pub struct UseSortedPropertiesOptions {
+    /// Partition the object properties by a newline.
+    pub partion_by_new_line: bool,
+    /// Partition the object properties by a comment.
+    pub partion_by_comment: bool,
+}
```

Comment:
As a general rule of thumb, we don't implement options for new rules on the first iteration to avoid unnecessary complexity. Lets leave these out for now.